include/dsn/dist/distributed_lock_service.h

```diff
@@ -69,7 +69,7 @@ namespace dsn
             /*
              * initialization routine
              */
-            virtual error_code initialize() = 0;
+            virtual error_code initialize(const char* work_dir, const char* lock_root) = 0;
```

Comment:
After all, the two services are not necessarily used together, so reading the shared config twice is acceptable. On the other hand, we can still share the ZK configs in config files, as you described in other thread - using [ZK] for shared config. using [svc] for provider-specific configs.